xlml/apis/test_config.py

```diff
@@ -183,6 +186,14 @@ def setup_script(self) -> Optional[str]:
   def test_script(self) -> str:
     return '\n'.join(self.run_model_cmds)
 
+  @property
+  def startup_script(self) -> str:
```

Comment:
I did it this way for the making the `metadata` code part of `create_queued_resource_request` easy and clean.

https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/84fa3aaaf9e9caf2ec026e6b76b5d6359ee99a1d/xlml/utils/tpu.py#L92

https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/84fa3aaaf9e9caf2ec026e6b76b5d6359ee99a1d/xlml/apis/test_config.py#L190-L195

Reply:
Hi @jonb377, yeah, I think this will be better.

Basically, we have a if statement inside of `create_queued_resource_request` ([here](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/utils/tpu.py#L109)), and generate the startup_script directly there from setup_cmd and run_model_cmd if `use_startup_script` is True.

Hi @tonyjohnchen, do you see any issue if we implement this way? So we won't need a startup_script method.

Reply:
@tonyjohnchen Sorry, I realize I put the comment in the wrong spot - my main concern is with tracking `use_startup_script` as part of the TestConfig. I think it would be better to keep TestConfig independent of the execution mode.

Let's sync up offline to see if this makes sense.

Reply:
Jon also mentioned `In short, I would say we should keep use_startup_script as a parameter to TpuQueuedResourceTask instead of attaching it to the TestConfig.`

Reply:
> Hi @jonb377, yeah, I think this will be better.
> 
> Basically, we have a if statement inside of `create_queued_resource_request` ([here](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/master/xlml/utils/tpu.py#L109)), and generate the startup_script directly there from setup_cmd and run_model_cmd if `use_startup_script` is True.
> 
> Hi @tonyjohnchen, do you see any issue if we implement this way? So we won't need a startup_script method.

Done with @RissyRan 's request.

Reply:
Discussed offline with Tony. I don't have a strong preference on this part. But would like to hear benefits vs. disadvantages of this change. Happy to discuss this together quickly.

Reply:
> Jon also mentioned `In short, I would say we should keep use_startup_script as a parameter to TpuQueuedResourceTask instead of attaching it to the TestConfig.`

This is possible, but it would be a lot of code changes. 

`run()` is using `provision >> run_model >> post_process >> clean_up`: https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/8f995e6cbaf5c0be211a75d92f0b7e23688d1876/xlml/apis/task.py#L76

`run_with_startup_script` is using `provision_with_startup_script >> post_process >> clean_up`: 
https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/8f995e6cbaf5c0be211a75d92f0b7e23688d1876/xlml/apis/task.py#L100

Both of them are under `TpuQueuedResourceTask`

For `post_process`, I will need to know `use_startup_script` to decide how to post metrics based on SSH or startup script:
https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/8f995e6cbaf5c0be211a75d92f0b7e23688d1876/xlml/utils/metric.py#L424

I could add a new section called `post_process_with_startup_script` and this would add a lot of new code.
I could also add a new parameter, and this wouldn't be too much different than hiding the `use_startup_script` param under `TestConfig`